[cutlass backend][ez] Ban FP32 output dtype from using CUTLASS GEMM backend

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #151279

FP32 not supported: https://github.com/pytorch/pytorch/issues/145952

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang committed:
[cutlass backend] Ban FP32 output dtype from using CUTLASS GEMM backend

reserved

Differential Revision: [D73005770](https://our.internmc.facebook.com/intern/diff/D73005770/)

[ghstack-poisoned]

ColinPeppler reviewed (changes requested):
I'm not sure banning FP32 is the right move here.

FP32 is supported by CUTLASS. My guess is the TF32 compilation issue is the result of some improper static typing.

henrylhtsang commented:
> I'm not sure banning FP32 is the right move here.
> 
> FP32 is supported by CUTLASS. My guess is the TF32 compilation issue is the result of some improper static typing.

I think its okay to ban for now. It is not a feature that has a lot of attention imo.

ColinPeppler reviewed (approved):
ok stamp to unblock, maybe add a test expecting fp32 to not be picked up by cutlass.

ColinPeppler commented:
or some logging, debug msg

henrylhtsang committed:
Update on "[cutlass backend][ez] Ban FP32 output dtype from using CUTLASS GEMM backend"


FP32 not supported: https://github.com/pytorch/pytorch/issues/145952

Differential Revision: [D73005770](https://our.internmc.facebook.com/intern/diff/D73005770/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot merge
